### PR TITLE
EVA-3317 - Accept webin token and provision FTP upload dir

### PR DESCRIPTION
**Pull request by sundarvenkata-EBI**

### CODE REVIEW

**Review by nitin-ebi — APPROVED** [review]

File: src/main/java/uk/ac/ebi/eva/submission/service/WebinTokenService.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        HttpEntity<String> entity = new HttpEntity<>("", headers);
+
+        RestTemplate restTemplate = new RestTemplate();
+        ResponseEntity<String> response = restTemplate.exchange(url, HttpMethod.GET, entity, String.class);
+        // Parse the response JSON to extract the new access token
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode responseJson = objectMapper.readTree(response.getBody());
+            return(responseJson.get("id").asText());
```
Comment (nitin-ebi) [inline src/main/java/uk/ac/ebi/eva/submission/service/WebinTokenService.java:30]: is there a reason the return string is wrapped in () ?

File: src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java [hunk 2]
```diff
@@ -6,34 +6,33 @@
...
-        String submissionId = createSubmissionDirectory();
+    public Submission initiateSubmission(String userId) {
+        String submissionId = UUID.randomUUID().toString();
+        String directoryToCreate = String.format("%s/%s", userId, submissionId);
+        globusDirectoryProvisioner.createSubmissionDirectory(directoryToCreate);
 
         Submission submission = new Submission(submissionId);
         submission.setStatus(SubmissionStatus.OPEN.toString());
         submission.setInitiationTime(LocalDateTime.now());
+        submission.setUploadUrl(uploadHttpDomain + "/" + directoryToCreate);
```
Comment (tcezard) [inline src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java:35]: I think we should also persist the userId

**Review by tcezard — APPROVED** [review]

File: src/main/java/uk/ac/ebi/eva/submission/service/WebinTokenService.java (hunk 1 above)
Comment (sundarvenkata-EBI) [reply src/main/java/uk/ac/ebi/eva/submission/service/WebinTokenService.java]: None :) Addressed.

File: src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java (hunk 2 above)
Comment (sundarvenkata-EBI) [reply src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java:36]: Addressed

**Pull request merged**